### PR TITLE
SelectControl : Deprecate 36px default size

### PR DESCRIPTION
**Pull request by PARTHVATALIYA**
Part of https://github.com/WordPress/gutenberg/issues/65751

## What?
Deprecate the 36px default size in SelectControl.

## Testing Instructions
- Unit tests pass
- Storybook stories should not log console warnings
- All code snippets in documentation (JSDoc, Storybook, README) should have the __next40pxDefaultSize prop enabled

### CODE REVIEW

File: packages/components/src/select-control/index.tsx [hunk 1]
```diff
@@ -94,6 +95,12 @@ function UnforwardedSelectControl< V extends string >(
 
 	const classes = clsx( 'components-select-control', className );
 
+	maybeWarnDeprecated36pxSize( {
+		componentName: 'SelectControl',
+		__next40pxDefaultSize,
+		size: undefined,
```
Comment (mirka) [inline packages/components/src/select-control/index.tsx]: This component does have a `size` prop so it needs to be passed.

```suggestion
		size,
```

File: packages/components/src/select-control/index.tsx [hunk 2]
```diff
@@ -154,6 +161,7 @@ function UnforwardedSelectControl< V extends string >(
  *
  *   return (
  *     <SelectControl
+ * 		 __next40pxDefaultSize
```
Comment (mirka) [inline packages/components/src/select-control/index.tsx]:
```suggestion
 *       __next40pxDefaultSize
 * ```

File: packages/components/src/select-control/stories/index.story.tsx [hunk 3]
```diff
@@ -65,6 +65,7 @@ const SelectControlWithState: StoryFn< typeof SelectControl > = ( props ) => {
 
 export const Default = SelectControlWithState.bind( {} );
 Default.args = {
```
Comment (mirka) [inline packages/components/src/select-control/stories/index.story.tsx:67]: The `WithCustomChildren` story also needs the prop.

**Review by mirka — COMMENTED** [review]
A couple more things:

- In addition to this, there are a lot of code snippets in Markdown files across the repo that use SelectControl, and all of them need the prop. Otherwise the devs referring to those snippets would be copy pasting deprecated code. (I added some clarification in the [original issue](https://github.com/WordPress/gutenberg/issues/65751)'s procedure checklist.)
- We should also add the prop to the [SelectControl usage](https://github.com/WordPress/gutenberg/blob/d57502ee6bad7bd6bf352a73700ba2e61d79c3ef/packages/components/src/disabled/stories/index.story.tsx#L56) in the Storybook story for the `Disabled` component.
- In a previous PR I [introduced](https://github.com/WordPress/gutenberg/pull/66747#discussion_r1847318146) a way to suppress warnings when the component's size is dependent on a parent component's size, and the parent component is responsible for logging the warning. I went ahead and added (70b67696b42fb6cae394ab3ac4e913d8145b8526) that to the relevant components here, which are DimensionControl and TreeSelect.

**Comment by mirka** [comment]
@PARTHVATALIYA Hi 👋 I need these merged soon, are you still available or do you need someone to take over? 🙂

**Comment by PARTHVATALIYA** [comment]
Hello @mirka, Sorry for the inconvenience! I wasn’t able to resolve the deprecation warning. Could you please guide me, or feel free to take over if needed?

**Review by mirka — APPROVED** [review]
No worries, thank you for your help here!

**Pull request merged**